bin/ruby-build

```diff
@@ -317,7 +323,6 @@ detect_http_client() {
       return
     fi
   done
-  echo "error: please install \`aria2c\`, \`curl\`, or \`wget\` and try again" >&2
```

Comment:
Sorry, but I don't understand this PR. We already display this error message on stderr. Why did you just move this elsewhere?

Reply:
I think it depends on file descriptor management.

I wrote some debug code on master branch to reproduce #1234  and investigate.

please see following code and please tell me when I was misunderstood.

## diff for debugging

```
# git diff
diff --git a/bin/ruby-build b/bin/ruby-build
index 5aba015..4fbf234 100755
--- a/bin/ruby-build
+++ b/bin/ruby-build
@@ -300,6 +300,7 @@ verify_checksum() {

 http() {
   local method="$1"
+  echo "### printing method and second argument: http_${method} $2 ###" # check arguments
   [ -n "$2" ] || return 1
   shift 1

@@ -310,6 +311,7 @@ http() {
 }

 detect_http_client() {
+  echo "## detect_http_client() is called ##"  # check whether detect_http_client is called
   local client
   for client in aria2c curl wget; do
     if type "$client" &>/dev/null; then
```

## output

requirements: now curl/wget/aria2c are absent

### when verbose option does not exist 

```
Downloading ruby-2.2.0.tar.bz2...
### printing method and second argument: http_head  ###
-> https://cache.ruby-lang.org/pub/ruby/2.2/ruby-2.2.0.tar.bz2
error: failed to download ruby-2.2.0.tar.bz2

BUILD FAILED (CentOS Linux 7 using ruby-build 20180822-8-g336584c)
```

I coudn't find error message on stderr.

### when verbose option exists. In other words, open the log file at fd4

```
# rbenv install 2.2.0 --verbose
/tmp/ruby-build.20180917212725.10142 ~/.rbenv/plugins/ruby-build
Downloading ruby-2.2.0.tar.bz2...
### printing method and second argument: http_head  ###
-> https://cache.ruby-lang.org/pub/ruby/2.2/ruby-2.2.0.tar.bz2
### printing method and second argument: http_get https://cache.ruby-lang.org/pub/ruby/2.2/ruby-2.2.0.tar.bz2 ###
error: please install `aria2c`, `curl`, or `wget` and try again
/root/.rbenv/plugins/ruby-build/bin/ruby-build: 行 310: http_get_## detect_http_client() is called ##: コマンドが見つかりません
error: failed to download ruby-2.2.0.tar.bz2

BUILD FAILED (CentOS Linux 7 using ruby-build 20180822-8-g336584c)
```

Reply:
> when verbose option does not exist

Do you not get in this case (on the master branch) the part of the output that says:

```
Inspect or clean up the working tree at /var/folders/5l/mnxly2_x6_s0pvh84q5qhsf80000gp/T/ruby-build.20180917131518.35519
Results logged to /var/folders/5l/mnxly2_x6_s0pvh84q5qhsf80000gp/T/ruby-build.20180917131518.35519.log

Last 10 log lines:
/var/folders/5l/mnxly2_x6_s0pvh84q5qhsf80000gp/T/ruby-build.20180917131518.35519 ~/p/ruby-build
error: please install `aria2c`, `curl`, or `wget` and try again
```

I do (bash 3.2.57 on macOS).

Reply:
I was investigated at #1201, fd 2 is not original stderr in `http` function.
So I think it should redirect to another fd or change redirects at `http get "$package_url" "$package_filename" >&4 2>&1`.